### ISSUE TITLE
Store Notices missing aria role attributes

### ISSUE THREAD

**Issue by sunyatasattva** [woothemes/woocommerce#60261]
### Describe the bug

Whilst working on #60253, I noticed that we have several templates used to create notices:

1. `templates/notices/` are used to render legacy notices in classic themes.
2. `templates/block-notices/` are used to render legacy notices in block themes.
3. `BlockTypes/StoreNotices.php` are the actual store notices blocks.

The latter of the three are missing any ARIA `role` attributes.

### Expected behavior

Notices should have appropriate `role` attribute.

### Actual behavior

`role` attributes are missing.

### Steps to reproduce

1. Go to the checkout page.
2. Trigger an error.
3. Notice that `role` attribute is missing from the HTML.

<img width="2190" height="1759" alt="Image" src="https://github.com/user-attachments/assets/cb91c950-4eac-42ea-b21c-094e4ae8fedc" />

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by RohithMacharla11** [comment]
I'm on it!

**Comment by sunyatasattva** [comment]
Alright, assigning this to you @RohithMacharla11 ! Feel free to tag me in the PR.

**Comment by RohithMacharla11** [comment]
Hi @sunyatasattva, I've updated the PR based on the bot's feedback. This now uses `role="status"` and includes the changelog file. Thank you for the opportunity!

**Comment by kkoteen** [comment]
Reviewed and confirmed this is still valid. Marking as ready for engineering, as none of the above PRs have resolved the issue.

**Comment by sunyatasattva** [comment]
Given no news from @RohithMacharla11 on his https://github.com/woocommerce/woocommerce/pull/60281 PR, I'm closing that PR and bringing this issue back to the drawing board.
